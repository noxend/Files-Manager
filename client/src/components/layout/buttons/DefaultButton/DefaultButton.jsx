import React from 'react';
import { Button } from 'antd';

const DefaultButton = (props) => {
  return (
    <Button type="default" {...props}>
      {props.children}
    </Button>
  );
};

export default DefaultButton;
