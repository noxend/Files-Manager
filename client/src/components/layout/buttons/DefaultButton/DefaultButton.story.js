import React from 'react';
import { storiesOf } from '@storybook/react';
import { withKnobs, text, boolean } from '@storybook/addon-knobs';

import DefaultButton from './DefaultButton';

const stories = storiesOf('Buttons', module);

stories.addDecorator(withKnobs);

stories.add('Default Button', () => {
  const name = text('Button Name', 'Some name');

  const loading = boolean('Loading', false);
  const disabled = boolean('Disabled', false);

  return (
    <DefaultButton loading={loading} disabled={disabled}>
      {name}
    </DefaultButton>
  );
});
