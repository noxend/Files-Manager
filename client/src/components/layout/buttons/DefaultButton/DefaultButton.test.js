import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import DefaultButton from './DefaultButton';

configure({ adapter: new Adapter() });

describe('Default Button', () => {
  it('Snapshot', () => {
    const tree = renderer.create(<DefaultButton />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('Props should to match the component', () => {
    const wrapper = shallow(<DefaultButton>Title</DefaultButton>);
    expect(wrapper.find('Button').prop('type')).toBe('default');
    expect(wrapper.find('Button').prop('children')).toBe('Title');
  });
});
