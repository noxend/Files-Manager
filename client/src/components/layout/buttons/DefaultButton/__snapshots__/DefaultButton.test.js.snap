// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Default Button Snapshot 1`] = `
<button
  className="ant-btn ant-btn-default"
  onClick={[Function]}
  type="button"
/>
`;
